OVERVIEW.md

```diff
@@ -0,0 +1,66 @@
+# Overview #
+
+Here it is described a brief overview of xdist's internal architecture.
+
+
+`xdist` works by spawning one or more **worker nodes**, which are controlled
+by the **master node**. Each **worker node** is responsible for performing 
+a full test collection and afterwards running tests as dictated by the **master node**.
+
+The execution flow is:
+
+1. **master node** spawns one or more **worker nodes** at the begginning of
+   the test session. The communication between **master** and **worker** nodes makes use of 
+   [execnet](http://codespeak.net/execnet/) and its [gateways](http://codespeak.net/execnet/basics.html#gateways-bootstrapping-python-interpreters).
+   The actual interpreters executing the code for the **worker nodes** might
+   be remote or local. 
+  
+1. Each **worker node** itself is a mini pytest runner. **workers** at this
+   point perform a full test collection, sending back the collected 
+   test-ids back to the **master node** which does not
+   perform any collection itself.
```

Comment:
That makes sense. I know from the Apache Spark project that object serialization can be a tough problem to solve.

This is off-topic as far as this review is concerned, but perhaps now is good time to revisit this issue now that libraries like [Dill](https://github.com/uqfoundation/dill) and [cloudpickle](https://github.com/cloudpipe/cloudpickle) exist. Perhaps it is not as difficult to solve as it once was.